### PR TITLE
More category tests closes #62

### LINKED ISSUE #62: Update Model Tests: Categorizations

### PR DESCRIPTION
**Pull request by bhargavisatpathy**
Added relationship test between category and events...created model Categorization to establish many to many relation between category and events.

### CODE REVIEW

File: spec/models/category_spec.rb [hunk 1]
```diff
@@ -14,6 +14,13 @@
...
 
-  xit "can have many events" do
+  it "can have many events" do
+    event1 = create(:event, title: "The Hobbit", description: "Hobbit movie 3")
+    event2 = create(:event, title: "Frozen", description: "movie about two sisters")
+
+    Categorization.create(category_id: category.id, event_id: event1.id)
+    Categorization.create(category_id: category.id, event_id: event2.id)
+    expect(category.events.count).to eq 2
+    expect(category.events.map(&:title)).to eq(["The Hobbit", "Frozen"])
```
Comment (KristaANelson) [inline spec/models/category_spec.rb]: I really like the concept of making two events and adding them to a category and testing that there are two. 

Is there a way to use the factory girl factories for this? 

  it "can have many events" do
    category = create(:category) #creates a standard category using the details from factories.rb
    event1 = create(:event) #creates a standard event from factories.rb
    event2 = create(:event, title: "Frozen", description: "movie about two sisters")
    #creates a second event with the only differance the title and description
    category.events << event1
    category.events << event2
    #adds the events to the cateogries
    expect(category.events.count).to eq 2
    expect(category.events.map(&:title)).to eq(["Blazers vs. Clippers", "Frozen"])
    #Blazers vs. Clippers is the default title setup in the factories file
  end

Comment (bhargavisatpathy) [inline spec/models/category_spec.rb]: updated category tests with factory girl...please review

**Comment by KristaANelson** [comment]
Updated the name sequence on factory.rb to all to be used for any table that has a name column. This way when you make one then one event, it'll automatically create it with a unique name. I think updated the tests to utilize this change. 

Since a may need to create a venue, I added on a validation for the venue name so that they don't keep making the same venue named the same thing over and over. 

Take a look and let me know if these updates make sense!

**Comment by KristaANelson** [comment]
I just pulled last request and merged it in locally and re-pushed to take care of the merge conflicts

File: spec/factories.rb [hunk 2]
```diff
@@ -1,13 +1,17 @@
 FactoryGirl.define do
 
   sequence :name do |n|
-    "category#{n}"
+    "name#{n}"
```
Comment (nathanows) [inline spec/factories.rb:4]: Like these sequence things a lot, didnt see these last time around :+1: :space_invader:

**Comment by nathanows** [comment]
looks good, nice work! :sake: :rabbit: :fallen_leaf:

**Pull request merged**